### PR TITLE
[DOC] Enhanced RDoc for #ln

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: lib/fileutils.rb [hunk 1]
```diff
@@ -358,25 +358,54 @@ def rmdir(list, parents: nil, noop: nil, verbose: nil)
...
-  #   FileUtils.ln(target,  dir, force: nil, noop: nil, verbose: nil)
-  #   FileUtils.ln(targets, dir, force: nil, noop: nil, verbose: nil)
+  # When +src+ is the path to an existing file or directory
+  # and +dest+ is the path to a non-existent file,
+  # creates a hard link at +dest+ pointing to +src+; returns zero:
+  #
+  #   Dir.children('tmp0/')                    # => ["t.txt"]
+  #   Dir.children('tmp1/')                    # => []
+  #   FileUtils.ln('tmp0/t.txt', 'tmp1/t.lnk') # => 0
+  #   Dir.children('tmp1/') # => ["t.lnk"]
```
Comment (peterzhu2118) [inline lib/fileutils.rb:368]: Should the output be aligned with the lines above?

File: lib/fileutils.rb [hunk 2]
```diff
@@ -358,25 +358,54 @@ def rmdir(list, parents: nil, noop: nil, verbose: nil)
...
+  #
+  # When +src+ is the path to an existing file or directory
```
Comment (peterzhu2118) [inline lib/fileutils.rb:370]: I don't think `src` can be a directory, it raises an error for me:

```ruby
irb(main):005:0> Dir.children('dir1')
=> ["test.txt"]
irb(main):006:0> Dir.children('dir2')
=> []
irb(main):007:0> FileUtils.ln('dir1', 'dir2')
/Users/peter/.rubies/ruby-3.1.2/lib/ruby/3.1.0/fileutils.rb:301:in `link': Operation not permitted @ rb_file_s_link - (dir1, dir2/dir1) (Errno::EPERM)
```

File: lib/fileutils.rb [hunk 3]
```diff
@@ -358,25 +358,54 @@ def rmdir(list, parents: nil, noop: nil, verbose: nil)
...
+  # and +dest+ is the path to an existing directory,
+  # creates a hard link at <tt>dest/src</tt> pointing to +src+; returns zero:
+  #
+  #   Dir.children('tmp2')               # => ["t.dat"]
+  #   Dir.children('tmp3')               # => []
+  #   FileUtils.ln('tmp2/t.dat', 'tmp3') # => 0
+  #   Dir.children('tmp3')               # => ["t.dat"]
+  #
+  # When +src+ is an array of paths to existing files or directories
```
Comment (peterzhu2118) [inline lib/fileutils.rb:379]: Just like above, I don't think `src` can contain paths of directories (only files).

File: lib/fileutils.rb (hunk 1 above)
Comment (BurdetteLamar) [reply lib/fileutils.rb]: Fixed.

File: lib/fileutils.rb (hunk 2 above)
Comment (BurdetteLamar) [reply lib/fileutils.rb]: Fixed.

File: lib/fileutils.rb (hunk 3 above)
Comment (BurdetteLamar) [reply lib/fileutils.rb]: Fixed.  (Actually there were three of these, all now fixed; hope that's correct.)

File: lib/fileutils.rb [hunk 4]
```diff
@@ -358,25 +358,54 @@ def rmdir(list, parents: nil, noop: nil, verbose: nil)
...
+  # creates a hard link at +dest+ pointing to +src+; returns zero:
+  #
+  #   Dir.children('tmp0/')                    # => ["t.txt"]
+  #   Dir.children('tmp1/')                    # => []
+  #   FileUtils.ln('tmp0/t.txt', 'tmp1/t.lnk') # => 0
+  #   Dir.children('tmp1/')                    # => ["t.lnk"]
+  #
+  # When +src+ is the path to an existing file
+  # and +dest+ is the path to an existing directory,
+  # creates a hard link at <tt>dest/src</tt> pointing to +src+; returns zero:
```
Comment (peterzhu2118) [inline lib/fileutils.rb:372]: This seems to suggest that it appends path of `src` to `dest`, which in this example:

```ruby
FileUtils.ln('tmp2/t.dat', 'tmp3')
```

I would expect the path to be `tmp3/tmp2/t.dat`.

```suggestion
  # creates a hard link in +dest+ pointing to +src+; returns zero:
```

File: lib/fileutils.rb [hunk 5]
```diff
@@ -358,25 +358,54 @@ def rmdir(list, parents: nil, noop: nil, verbose: nil)
...
+  #
+  #   Dir.children('tmp2')               # => ["t.dat"]
+  #   Dir.children('tmp3')               # => []
+  #   FileUtils.ln('tmp2/t.dat', 'tmp3') # => 0
+  #   Dir.children('tmp3')               # => ["t.dat"]
+  #
+  # When +src+ is an array of paths to existing files
+  # and +dest+ is the path to an existing directory,
+  # then for each path +target+ in +src+,
+  # creates a hard link at <tt>dest/target</tt> pointing to +target+;
```
Comment (peterzhu2118) [inline lib/fileutils.rb:382]: Ditto:

```suggestion
  # creates a hard link in +dest+ pointing to +target+;
```

**Review by peterzhu2118 — APPROVED** [review]
Looks good!

**Pull request merged**